### ISSUE TITLE
Adding temperature and humidity compensation

### ISSUE THREAD

**Issue by Cinezaster** [mdsiraj1992/Gassensors#3]
When I see the documentation of all the sensors I see a temperature/humidity compensation graph. 
If your using these sensors outside, compensation would be a good thing.

**Comment by mdsiraj1992** [comment]
Yes compensation is very much required but i have not yet understood how to do that.
I am open to suggestions or help.

**Comment by Cinezaster** [comment]
I'm going to test this library over the next days. Needed to burn in my sensors, first.
I'm having MQ-02, MQ-04 and MQ-09.
Happy to look into this issue.

**Comment by saberjam** [comment]
Hi has anyone know or have MQ codes that are correlated already with temperature and humidity?

**Comment by Cinezaster** [comment]
I did research this and came to the conclusion that these sensors need to be calibrated against a reference gas, temperature and humidity. It is hard to get correct data based on this code although it is the beste effort to have a out of the box functions that generate a number that is somehow correlated to real world values. 
I abandoned this road of using the MQ sensors. I'm using a setup with reference gas and added temperature and humidity sensors. To get data I'm using post processing of the raw output of the sensors and I calibrate my data to the measurements of the reference gas.

**Comment by zoomx** [comment]
The only calibration data that are available are the ones published on datasheet.

**Comment by Cinezaster** [comment]
@zoomx if it would be calibration data you would receive a unique dataset for every sensor, when you buy it. There are to much outside factors that influence the output of these sensors.
Also, these sensors have a operation live of about 2 year, if they are used properly. 
If they are used out of specification it is worse. Or if you would turn them on or off (= heating / cooling) frequently they degrade even faster, because of oxidation of the sensor.

The data on the sheet is more or less the specification of the sensor.
These sensors are to be used in alarm systems where they would trigger alarms when certain gasses are above a certain threshold. 
But as I stated before, if you calibrate them against a know reference gas, it is possible to use them for sensing.
Look at [this e nose example](http://www.maskau.dk/projects/electronic-nose) from Michael Madsen, he uses the raw data from multiple sensors and does some neat post processing on it to classify different agents.

**Comment by zoomx** [comment]
I know. When possible I use NDIR sensors but they too need calibration.
Calibrating MQ sensors is a challenge since they are sensible to many gases. But the use of all MQ together is a good idea! Thanks!
